CIP-XXXX/CIP-XXXX.md

```diff
@@ -0,0 +1,180 @@
+
+#### cardano.{walletName}.name: String
+
+A name for the wallet which can be used inside of the dApp for the purpose of asking the user which wallet they would like to connect with.
+
+#### cardano.{walletName}.icon: String
+
+A URI image (e.g. data URI base64 or other) for img src for the wallet which can be used inside of the dApp for the purpose of asking the user which wallet they would like to connect with.
+
+### Full API
+
+Upon successful connection via `cardano.{walletName}.enable()`, a javascript object we will refer to as `API` (type) / `api` (instance) is returned to the dApp with the following method.
+
+This is where extensions are free to add functionality as they see fit. Furthermore extensions are free to include additional sections, naming their APIs as they see fit, an example could be an `Experimental API`.
+
+#### api.getExtensions(): Promise\<Extension[]>
+
+Retrieves the list of extensions enabled by the wallet. This may be influenced by the set of extensions requested in the initial `enable` request.
+
+## Rationale: how does this CIP achieve its goals?
+
+This API does not offer *useful* functionality for dApp-Wallet interactions; rather, all useful functionality should be offered through extensions. By keeping this API as a skeleton, we aim to avoid the chance of introducing functionality we come to regret later. Instead, this proposal should be the minimum foundation required to build on top of.  
+
+The alternative to this proposal is the addition of extensions to CIP-30's specification. This would mean that the CIP-30 API would be included in all extended dApp-Wallet connections. With the current landscape of dApps and wallets this seems fine, but it is uncertain if we want to indefinitely support the CIP-30 API.
+
+By stripping functionality from this API we facilitate dApp-Wallet connections with less functions than CIP-30 API. A use case for this is connections where minimal wallet information is wished to be shared with a dApp.
+
+Here we attempt to mitigate the impact of succeeding CIP-30 as *the* Cardano wallet connector, by enabling it's functionality by default at connection time. This should minimize the impact this proposal could have on established dApps who support CIP-30. By only adding what is necessary for extensions we aim to reduce the required load for wallets to adopt this standard.
+
+### Extensions
+
+Extensions provide an extensibility mechanism and a way to negotiate (possibly conflicting) functionality between a dApp and a wallet provider. There's rules enforced as for what extensions a wallet decide to support or enable. The current mechanism only gives a way for wallets to communicate their choice back to a dApp.
+
+We use object as extensions for now to leave room for adding fields in the future without breaking all existing interfaces. At this point in time however, objects are expected to be singleton.

```

Comment:
Don't quite understand what 'singleton' means in this context (or what is the 'object'). Would be nice to elaborate.

Reply:
This means that `extension` objects cannot pass additional fields at enable time, currently.

Reply:
Do we need to limit this in this text? Isn't it enough to say that this CIP only defines a single key in the object passed, and that it's up to Extensions to define additional fields in their respective CIP, if any?

Reply:
> This means that extension objects cannot pass additional fields at enable time, currently

Thanks. Could be just me not understanding it from text, but consider rewording it @Ryun1 

> Do we need to limit this in this text? Isn't it enough to say that this CIP only defines a single key in the object passed, and that it's up to Extensions to define additional fields in their respective CIP, if any?

The simpler the base dApp connector cip the better imho. The scope of the extension CIPs can then be contained to just the API object that `enable()` returns. The extensions can have API methods (or arguments for those methods) instead. If we allow configuration at `enable()` time then we can expect to see extension designs that would require the dApp calling `enable` multiple times with different arguments. Do we want that? :sweat_smile:

Reply:
> Do we want that?
You raise a good point, I think this could become messy quickly.

And I have removed this sentence to avoid the confusing wording.